### ISSUE TITLE
program aggregated does not compile

### ISSUE THREAD

**Issue by darlentar** [rixed/ramen#1037]
Currently, aggregated.ramen program does not compile.

`ramen compile -r examples/programs/monitoring/generated/aggregated.ramen`

```
18h58m49: Server MD5 for examples/programs/monitoring/generated/aggregated is "4d6b4f81a85b46b85e6a4bd6df140136" instead of "d47f7493af36948f868d47ac80ea5e85", waiting...
18h58m50: SourceInfo { src_ext:"ramen", md5:"d47f7493af36948f868d47ac80ea5e85", err:"Error at line 3, col 53 (near ')'): Cannot find \";\" > blanks > error response > unsat response > response > SMT output" }
18h58m50: Raised by primitive operation at file "src/RamenCliCmd.ml", line 412, characters 69-85
Called from file "src/RamenSyncZMQClient.ml", line 478, characters 10-44
Called from file "src/RamenSyncZMQClient.ml", line 485, characters 4-26
Called from file "src/RamenHelpers.ml", line 275, characters 6-10
Called from file "src/RamenHelpers.ml", line 17, characters 8-11
Called from file "src/RamenHelpers.ml", line 17, characters 8-11
Called from file "src/RamenSyncHelpers.ml", line 41, characters 2-261
Called from file "src/RamenCliCmd.ml", line 425, characters 2-1023
Called from file "list.ml", line 106, characters 12-15
```

But with a small change, it compile:
```

-- vim: ft=sql expandtab spell spelllang=en

-- Basic time aggregates of HTTP logs.
-- The coarser the retention and the longer we expect this to be available in
-- the future.

PARAMETERS
  time_period {seconds} DEFAULTS TO 1min,
  clock_jitter {seconds} DEFAULTS TO 1s;

DEFINE PERSIST FOR 1h http AS
  FROM ../logs/http
  -- start, stop, request_size response_size, method, code, url
  SELECT
    -- Times
    MIN start,
    MAX stop,
    stop - start AS _rt,
    TRUNCATE(start, time_period) AS start,
    start + time_period AS stop,

    -- The key
    SUM 1 AS count,
    method,
    code,
    SUBSTRING(url, 0, 10) AS url,

    MIN _rt,
    MAX _rt,
    AVG _rt,
    [95th; 99th] PERCENTILE (SAMPLE LOCALLY(1000, _rt)) AS rt_perc,
    -- Until rmadmin field selection can handle compound types:

    MIN request_size,
    get(0, rt_perc) AS rt_perc_95th,
    MAX request_size,
    AVG request_size,
    [95th; 99th] PERCENTILE (SAMPLE LOCALLY(1000, request_size)) AS req_size_perc,

    MIN response_size,
    MAX response_size,
    AVG response_size,
    [96th; 99th] PERCENTILE (SAMPLE LOCALLY(1000, response_size)) AS resp_size_perc

  GROUP BY
    start // time_period, method, code, SUBSTRING(url, 0, 10)
  COMMIT AFTER
    in.start > out.stop + 2 * clock_jitter;
```

Note that `get(0, rt_perc) AS rt_perc_95th` was moved two lines away.

**Comment by darlentar** [comment]
bisect gives commit: #c9c6ee8c340eedafde22f6e54288b5992ae28035

**Comment by rixed** [comment]
Works for me on master:
```
$ ramen compile -L examples/programs/monitoring/generated/ examples/programs/monitoring/generated/aggregated.ramen
21h40m24: Parsing program aggregated
21h40m24: Typing program aggregated
21h40m25: Compiling program aggregated
21h40m25: Compiling "examples/programs/monitoring/generated/aggregated_params_v66.ml"
21h40m25: Compiling "examples/programs/monitoring/generated/aggregated_7361e59dde7761ff0e928c5999631fd6_v66.ml"
21h40m25: Linking "examples/programs/monitoring/generated/aggregated_casing_v66.ml"
```

Also works with 8da8587.

Are you going through a confserver? If so, have you taken that into account when bisecting?

**Comment by darlentar** [comment]
Indeed it works (now) on master. But my remarks are still valid.

`~/github/ramen$ git bisect start master 5647f9d3de3b837ccf6f280243c776f8dbdd2f7e`
`~/github/ramen$ git bisect run /tmp/bisect.sh`
```
~/github/ramen$ cat /tmp/bisect.sh 
#!/usr/bin/bash
./config.status Makefile
make clean
make -j10 src/ramen
pkill -f ramen
rm -rf /tmp/ramen
ramen confserver --insecure 29340 --debug & # --no-examples
ramen compserver --confserver=localhost:29340 --debug --daemonize
ramen supervisor --confserver=localhost:29340 --debug --daemonize
ramen choreographer --confserver=localhost:29340 --debug --daemonize
ramen compile -L examples/programs/monitoring/generated/ examples/programs/monitoring/generated/aggregated.ramen
if [ $? -eq 0 ]
then
  exit 1
else
  exit 0
fi
```

Output:
```
16a010c30f23dcfaf9490e699dca1706c1032d8b is the first new commit
commit 16a010c30f23dcfaf9490e699dca1706c1032d8b
Author: Damien Bihel <dbihel@accedian.com>
Date:   Fri Oct 11 18:30:03 2019 +0200

    add: add Char type

:100644 100644 c865cea9b86cf46392443a2954b12a0770c8b04d 903cd4a2ba8fdac384a31b2621288351bc0d058c M	Makefile.in
:040000 040000 6aaac789b833e2f0dc607f3cce0d5fa9e8500b3b 1288fc6a8e5fdefd73e37ae0400d19b6feada580 M	src
:040000 040000 98075d929ddb5a3b45b14ba78701e95304103442 065cef1224fdd9b0f8b9b54eba41892e4c8a32cb M	tests
bisect run success
```

=================
`~/github/ramen$ git bisect start 552eda8be835a6e5bb9c553e751247e0ef0fcdbc 974379eb18c5d44ffa7bb2ca9c58d027bd5d1ac6`
`~/github/ramen$ git bisect run /tmp/bisect.sh`
```
~/github/ramen$ cat /tmp/bisect.sh 
#!/usr/bin/bash
./config.status Makefile
make clean
make -j10 src/ramen
pkill -f ramen
rm -rf /tmp/ramen
ramen confserver --insecure 29340 --debug & # --no-examples
ramen compserver --confserver=localhost:29340 --debug --daemonize
ramen supervisor --confserver=localhost:29340 --debug --daemonize
ramen choreographer --confserver=localhost:29340 --debug --daemonize
ramen compile -L examples/programs/monitoring/generated/ examples/programs/monitoring/generated/aggregated.ramen
```

Output
```
11h11m40: Cannot parse solver output:
sat
(error "line 1043 column 16: unsat core construction is not enabled, use command (set-option :produce-unsat-cores true)")
(error "line 1044 column 10: model is not available")

Error at line 3, col 53 (near ')'): Cannot find ";" > blanks > error response > unsat response > response > SMT output
8da8587949c089faeef5c85f910c54a38d420f91 is the first new commit
commit 8da8587949c089faeef5c85f910c54a38d420f91
Author: Cedric Cellier <rixed@happyleptic.org>
Date:   Wed Nov 6 11:41:27 2019 +0100

    Alternate fix for imaginary records: type invalid fields out of existence
    
    Create a special sort of values for field indices so that Z3 can no longer
    invent invalid records.
    
    As a welcomed side effect, make typing of some complex programs almost 10
    times faster :)
    
    Closes #1030

:040000 040000 072882fd8e922db27a1537f62d454e376a5c3f6f 9f8c5b1a0e28726a4c541a0a811abfff7962db37 M	src
bisect run success
```

**Comment by rixed** [comment]
The output you got at 8da858 matches the erroneous result I got at some point as well and that was fixed by 50cdbe7c904bb.
